solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -868,7 +869,11 @@ public ClusterState getClusterState() {
   }
 
   public Object getUpdateLock() {
-    return this;
+    return updateLock;
```

Comment:
Scope is becoming slightly more than a refactor; now getUpdateLock has a dedicated object.  Was that necessary to achieve the refactoring aims?  For example it creates new problems... note that `createClusterStateWatchersAndUpdate()` is marked synchronized, thus it synchronizes on "this" but now it's different than what getUpdateLock returns.

Reply:
You are right. Fixed this